qiskit/circuit/library/n_local/n_local.py

```diff
@@ -934,8 +948,21 @@ def _build(self) -> None:
             return
 
         # use the initial state circuit if it is not None
+        # if the initail_state is an instance of QuantumCircuit, do not call construct_circuit
         if self._initial_state:
-            circuit = self._initial_state.construct_circuit("circuit", register=self.qregs[0])
+            if isinstance(self._initial_state, QuantumCircuit):
+                circuit = self._initial_state.copy()
+            else:
+                warnings.warn(
```

Comment:
I think it should be enough to warn only in the setter, since the code will always pass through the setter if an initial state is supplied 🙂